src/v/cluster/types.cc

```diff
@@ -478,6 +478,26 @@ adl<cluster::create_topics_request>::from(iobuf_parser& in) {
     return cluster::create_topics_request{std::move(configs), timeout};
 }
 
+void adl<cluster::create_non_replicable_topics_request>::to(
+  iobuf& out, cluster::create_non_replicable_topics_request&& r) {
+    reflection::serialize(out, std::move(r.topics), r.timeout);
+}
+
+cluster::create_non_replicable_topics_request
+adl<cluster::create_non_replicable_topics_request>::from(iobuf io) {
```

Comment:
Do you need these two different overloads of `adl<>::from()` implemented? I think you would only need the one that accepts an `iobuf_parser&`